examples/golang/docker-compose.yaml

```diff
@@ -11,6 +11,12 @@ services:
         condition: service_healthy
     ports:
     - "${PORT_PROXY:-10000}:10000"
+    environment:
+    # The Envoy Golang filter is designed to be run with the GODEBUG=cgocheck=0
+    # environment variable set.
+    # This disables the cgo pointer check.
+    # Failure to set this environment variable will cause Envoy to crash!
```

Comment:
is this still the case?

i think this is really messy and should be addressed properly

im also trying to understand why this needs to change here now when it worked previously

Reply:
> environment variable will cause Envoy to crash!

this just _should not happen_

Reply:
1. this is an example, for people easy to copy & paste.
2. yep, we still need it, there is a proposal for golang, to hard code this ENV in the envoy Go SDK, but it may wait for the next Go release cycle, since the current Go release cycle is freezed, new changes won't be accepted.
https://github.com/golang/go/issues/60426
3. I believe we could fix it eventually, but may still need some time, just some compromise for now.

Reply:
> hard code this ENV in the envoy Go SDK, but it may wait for the next Go release cycle

im struggling to understand why we need to wait for a Go release to address this

surely we can just set/change this ourselves in the env (ie in envoy runtime) if it is necessary before loading the filter

Reply:
yep, it's a bit complicated.
actually, when we set a env in C/C++, glibc is updating a copy of a env list.
but, when open a dynamic so, glibc use the original env list, that means the golang so, won't see the new env list.

and, the Go runtime doesn't honor the changes, by using the Go side Set env API.
So, our plan is, let the Go runtime support updating it by Go side Set env API, so that we can set it in the Envoy Go sdk, then, people do not need to care about it, eventually.